cli/best_ancestor.go

```diff
@@ -0,0 +1,105 @@
+package cli
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+
+	"github.com/yourbase/yb/plumbing/log"
+)
+
+func fastFindAncestor(r *git.Repository) (h plumbing.Hash, c int, branchName string) {
+	/*
+		go doc gopkg.in/src-d/go-git.v4/plumbing/object Commit.MergeBase
+
+		func (c *Commit) MergeBase(other *Commit) ([]*Commit, error)
+		MergeBase mimics the behavior of `git merge-base actual other`, returning
+		the best common ancestor between the actual and the passed one. The best
+		common ancestors can not be reached from other common ancestors.
+	*/
+	c = -1
+
+	ref, err := r.Head()
+	if err != nil {
+		log.Errorf("%v", err)
+		return
+	}
+
+	remoteBranch := findRemoteBranch(ref, r)
```

Comment:
We talked about this outside of this diff -- there's a minor issue with multiple remotes so we should only use github ones for the moment. Long term we should probably negotiate a remote that's shared between the user and the CI system (i.e what the github hack will do but generalized)